### PR TITLE
Show error page or user notification in case of exceptions

### PR DESCRIPTION
**Pull request by KochTobi**
- [x] Make ApplicationException explicit and remove extending exceptions
- [x] Add a handler for all exceptions occurring during the lifetime of a UI
- [x] Add handlers to deal with routing exceptions
- [x] Provide easy way to extend error pages more specifically

### CODE REVIEW

**Comment by KochTobi** [comment]
includes changes from #191

File: commons/src/main/java/life/qbic/application/commons/ApplicationException.java [hunk 1]
```diff
@@ -130,6 +130,29 @@ public ErrorParameters errorParameters() {
...
+  public static ApplicationException wrapping(String message, Throwable e) {
+    if (e instanceof ApplicationException applicationException) {
+      return new ApplicationException(message, e, applicationException.errorCode(),
+          applicationException.errorParameters());
+    } else {
+      return new ApplicationException(message, e);
+    }
+  }
+
+  public static ApplicationException wrapping(Throwable e) {
```
Comment (sven1103) [inline commons/src/main/java/life/qbic/application/commons/ApplicationException.java:142]: Some documentation on the API to make the intention clear :)

File: commons/src/main/java/life/qbic/application/commons/ApplicationException.java [hunk 2]
```diff
@@ -130,6 +130,29 @@ public ErrorParameters errorParameters() {
     return errorParameters;
   }
 
+  public static ApplicationException wrapping(String message, Throwable e) {
```
Comment (sven1103) [inline commons/src/main/java/life/qbic/application/commons/ApplicationException.java:133]: Documentation missing

File: commons/src/main/java/life/qbic/application/commons/ApplicationException.java [hunk 3]
```diff
@@ -130,6 +130,29 @@ public ErrorParameters errorParameters() {
...
+  public static ApplicationException wrapping(Throwable e) {
+    if (e instanceof ApplicationException applicationException) {
+      return new ApplicationException(e, applicationException.errorCode(),
+          applicationException.errorParameters());
+    } else {
+      return new ApplicationException(e.getMessage(), e);
+    }
+  }
+
+  public static <E extends ApplicationException> ApplicationException wrapping(String message,
```
Comment (sven1103) [inline commons/src/main/java/life/qbic/application/commons/ApplicationException.java:151]: Missing docs

File: vaadinfrontend/src/main/java/life/qbic/datamanager/exceptionhandling/ErrorMessageTranslationService.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import org.springframework.stereotype.Service;
+
+/**
+ * <b>short description</b>
+ * TODO
+ * <p>detailed description</p>
+ *
+ * @since <version tag>
+ */
+@Service
```
Comment (sven1103) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/exceptionhandling/ErrorMessageTranslationService.java:20]: Docs missing :)

File: vaadinfrontend/src/main/java/life/qbic/datamanager/exceptionhandling/UiExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import life.qbic.application.commons.ApplicationException;
+import life.qbic.datamanager.exceptionhandling.ErrorMessageTranslationService.UserFriendlyErrorMessage;
+import life.qbic.datamanager.views.notifications.ErrorMessage;
+import life.qbic.datamanager.views.notifications.StyledNotification;
+import life.qbic.logging.api.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UiExceptionHandler {
```
Comment (sven1103) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/exceptionhandling/UiExceptionHandler.java:17]: Java Docs missing

File: vaadinfrontend/src/main/java/life/qbic/datamanager/exceptionhandling/routing/exception/ExceptionErrorPage.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * <b>short description</b>
+ * TODO
+ * <p>detailed description</p>
+ *
+ * @since <version tag>
+ */
+@AnonymousAllowed
+public class ExceptionErrorPage extends VerticalLayout implements ErrorPage<Exception> {
```
Comment (sven1103) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/exceptionhandling/routing/exception/ExceptionErrorPage.java:21]: Java docs

**Review by sven1103 — CHANGES_REQUESTED** [review]
Only documentation on classes and API missing, the implementation looks good !

**Review by sven1103 — APPROVED** [review]
LGTM 🚀

**Pull request merged**